javascript-carousel/main.js

```diff
@@ -0,0 +1,91 @@
+var $circleButtons = document.querySelectorAll('.circle');
+var $viewList = document.querySelectorAll('.view');
+var $rightArrow = document.querySelector('.right');
+var $leftArrow = document.querySelector('.left');
+var $bottomButton = document.querySelectorAll('.bottom');
+var $navBar = document.querySelector('.nav-bar');
+var timer = setInterval(changeImage, 3000);
+$rightArrow.addEventListener('click', changeImageWithRightArrow);
+$leftArrow.addEventListener('click', changeImageWithLeftArrow);
+$navBar.addEventListener('click', changeButton);
+
+function changeImage() {
+  var i = 0;
+  for (i; i < $viewList.length; i++) {
+    if ($viewList[i].className === 'view') {
+      $viewList[i].className = 'view hidden';
+      $circleButtons[i].className = ('circle far fa-circle fa-3x');
+      i++;
+      if (i < $viewList.length) {
+        $viewList[i].className = 'view';
+        $circleButtons[i].className = 'circle fas fa-circle fa-3x';
+      } else if (i === $viewList.length) {
+        $viewList[0].className = 'view';
+        $circleButtons[0].className = 'circle fas fa-circle fa-3x';
+      }
```

Comment:
This code is repeated quite a bit. See if you can define a function that takes a new index and updates the DOM accordingly. Then, re-use that function.

```js
function activateIndex(index) {

}
```